### PR TITLE
flagd grps/rest interface .proto files and buf definitions

### PR DESCRIPTION
**Pull request by James-Milligan**
Definitions are required for implementation of the grpc/rest interface for flagd. The buf files and repo structure are subject to change when a shift is made to cue.

Signed-off-by: James-Milligan <james@omnant.co.uk>

### CODE REVIEW

**Comment by AlexsJones** [comment]
Maybe we start subfoldering to keep this project tidy
e.g.

```
-openapi
-protobuf
-json
  - v1
```

Then you could put all the buf dependencies in there rather than project root

File: protobuf/v1/schema.proto [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+    string variant = 3;
+}
+
+message ResolveNumberRequest {
+    string flag_key = 1;
+    google.protobuf.Struct context = 2;
+    float default_value = 3;
+}
+message ResolveNumberResponse {
+    float value = 1;
```
Comment (toddbaert) [inline protobuf/v1/schema.proto:39]: Should we consider oneOf here to support ints and floats? Not sure it's 100% needed, float may be good enough... something to consider.

File: protobuf/v1/schema.proto [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+            body: "*"
+        };
+        option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+            summary: "Resolve boolean value for flagKey";
+            responses: {
+                key: "200"
+                value: {
+                    description: "OK"
+                    schema: {
+                        example: '{"flag_key":true,"reason":"","variant":""}';
```
Comment (beeme1mr) [inline protobuf/v1/schema.proto:68]: Should `flag_key` actually be value in this example?

**Review by toddbaert — APPROVED** [review]
LGTM, but I would correct the example as @beeme1mr mentioned.

File: protobuf/v1/schema.proto (hunk 2 above)
Comment (James-Milligan) [reply protobuf/v1/schema.proto]: these have been removed now, originally these were for padding out the opapiv2 docs, this will no longer be the case

**Pull request merged**